app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -334,13 +334,14 @@ private void addRoomColumns(
         adapterByRoomIndex.clear();
 
         int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
-        LecturesByRoomIndex lecturesByRoomIndex = new LecturesByRoomIndex(lectures);
         LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
 
-        for (int roomIndex = 0; roomIndex < roomCount; roomIndex++) {
-            List<Lecture> roomLectures = lecturesByRoomIndex.get(roomIndex);
+        List<RoomData> roomDataList = scheduleData.getRoomDataList();
```

Comment:
I would prefer a for of / .forEach loop here using the iterability of List.

Reply:
`roomIndex` is used further down in the loop body.